skfda/preprocessing/dim_reduction/_pace.py

```diff
@@ -0,0 +1,1441 @@
+    coeff = 1 / ((2 * np.pi) ** (n_dims / 2))
+    return np.array(coeff * np.exp(-0.5 * norm_sq))
+
+
+class PACE(  # noqa: WPS230
+    InductiveTransformerMixin[FData, NDArrayFloat, object],
+    BaseEstimator,
+):
+    r"""
+    FPCA through conditional expectation.
+
+    Class that implements functional principal component analysis through
+    conditional expectation. This method is native to FDataIrregular.
+
+    For more information about the theoretical foundation for this algorithm,
+    see: footcite:t:`yao+muller+wang_2005_pace`.
+
+    Parameters:
+        n_components: If parameter is an integer, it refers to the number of
+            principal components to keep from functional principal component
+            analysis. If parameter is a float in the range (0.0, 1.0), it
+            refers to the minimum proportion of variance explained by the
+            selected principal components. Defaults to None (maximum number
+            of principal components that can be extracted).
+        assume_noisy: Set to ``False`` when the data is assumed to be
+            noiseless. Otherwise, when smoothing the covariance surface, the
+            diagonal will be treated separately. Defaults to ``True``.
+        kernel_mean: callable vectorized univariate smoothing kernel function
+            for the mean, of the form :math:`K(t)`, where `t` are the
+            n-dimensional time point, with n being the dimension of the domain.
+            Defaults to a Gaussian kernel.
+        bandwidth_mean: bandwidth to use in the smoothing kernel for the mean.
+            If a float is given, it is used as the bandwidth. If a tuple is
+            given, it is used as the bandwidth search range, and the bandwidth
+            is calculated using the GCV method.
+        kernel_cov: callable vectorized univariate smoothing kernel function
+            for the covariance and calculations regarding its diagonal. It
+            should have the form :math:`K(t)`, where `t` are the
+            n-dimensional time point, with n being the dimension of the domain.
+            To smooth the covariance, each value in the two directions will be
+            calculated with the function and the two values will be multiplied,
+            acting as an isotropic kernel. Defaults to a Gaussian kernel.
+        bandwidth_cov: bandwidth to use in the smoothing kernel for the
+            covariance. If a float is given, it is used as the bandwidth. If a
+            tuple is given, it is used as the bandwidth search range, and the
+            bandwidth is calculated using the GCV method.
+        bw_cov_n_grid_points: number of grid points to calculate the bandwidth
+            for the covariance. This parameter's main purpose is to reduce the
+            computational cost of the GCV method. If the parameter
+            ``bandwidth_cov`` is provided as a float, this parameter is
+            ignored. Defaults to 30.
```

Comment:
Is it wise that the default is not the full grid? What is the impact in performance?